modules/tinymce/src/themes/silver/main/ts/modes/Iframe.ts

```diff
@@ -74,8 +74,11 @@ const render = (editor: Editor, uiComponents: RenderUiComponents, rawUiConfig: R
 
   loadIframeSkin(editor);
 
-  Attachment.attachSystemAfter(Element.fromDom(args.targetNode), uiComponents.mothership);
-  Attachment.attachSystem(Body.body(), uiComponents.uiMothership);
+  const eTargetNode = Element.fromDom(args.targetNode);
+  const uiRoot = ShadowDom.getContentContainer(ShadowDom.getRootNode(eTargetNode));
+
+  Attachment.attachSystemAfter(eTargetNode, uiComponents.mothership);
+  Attachment.attachSystem(uiRoot, uiComponents.uiMothership);
```

Comment:
I'm concerned this will cause a number of positioning issues, as I wouldn't be surprised if a bunch of code is relying on it being in the root document body, but I guess we can deal with them when they pop up?

Reply:
Yeah, I expect lots of edge cases. For now, I'm just trying to make improvements as I go, while not breaking non-shadow behaviour.

With this PR and the next one, we should be at the stage where there's enough working that we can start flushing these issues out.